lineman/app/components/thread_page/comment_form/comment_form.coffee

```diff
@@ -3,15 +3,16 @@ angular.module('loomioApp').directive 'commentForm', ->
   restrict: 'E'
   templateUrl: 'generated/components/thread_page/comment_form/comment_form.html'
   replace: true
-  controller: ($scope, FlashService, Records, CurrentUser) ->
+  controller: ($scope, FormService, Records, CurrentUser) ->
     group = $scope.comment.discussion().group()
     discussion = $scope.comment.discussion()
 
-    $scope.submit = ->
-      $scope.comment.save().then ->
-        $scope.comment = Records.comments.build(discussion_id: discussion.id)
-        $scope.$emit('commentSaveSuccess')
-        FlashService.success('comment_form.messages.created')
+    $scope.submit = FormService.submit $scope, $scope.comment,
+      flashSuccess: 'comment_form.messages.created'
+      successCallback: =>
+        $scope.comment.body = ''
```

Comment:
I think it's a bit weird/wrong choosing to reset fields rather than building a new comment. It could lead to problems in the future.

It's only a side effect of the current create process that the actual comment object from the form is not also the final comment object when saved. When you update a comment record we clone it for this very reason. Are you ok with returning to the `Records.comments.build` method of reset?